velox/core/QueryConfig.h

```diff
@@ -338,6 +338,10 @@ class QueryConfig {
   static constexpr const char* kEnableExpressionEvaluationCache =
       "enable_expression_evaluation_cache";
 
+  // Timestamp unit used during Velox-Arrow conversion.
+  static constexpr const char* kArrowBridgeTimestampUnit =
```

Comment:
it's a query level config? But what if we have two timestamp fields with different precision?

Reply:
Yes, different units among fields is not supported.

Reply:
This is already an improvement over the previous situation. Are we considering going a step further to support configuration at the field level? However, that would entail significant changes, as it would require introducing a definition of timestamp precision in Velox's type system, along with the corresponding type inference.

Reply:
> Are we considering going a step further to support configuration at the field level?

We are testing with Spark, and at least so far no corresponding usage scenarios have been found. Spark provides configurable unit for writing and reading timstamps in Parquet, but it is context level and does not vary among fields. I'm wondering which system you are testing, and how common this issue is. cc @Yuhta

Reply:
I have an idea, we can pass through the expected arrow shcema to  exportToArrow function, and exporting timestamp according to the expected unit of the timestamp field. @rui-mo what do you think about

Reply:
As velox do not take responsibility for building query plan, the framework invoking velox must know what the output schema of velox task is. In this way, we can specify different timestamp unit across columns.

Reply:
> the framework invoking velox must know what the output schema of velox task is

I see your point. I assume the gap is Timestamp type in Velox does not hold a `precision` or `unit` field, so the caller cannot specify that information in the output schema. To achieve a differentiated timestamp unit when exporting to arrow, Velox vector or type needs to hold relating information I think.

Reply:
One method I can think of is to pass through a ArrowSchema, like:
``` c++
void exportToArrow(
    const VectorPtr& vec,
    ArrowSchema& arrowSchema,
    const ArrowOptions& options,
    const ArrowSchema* expectedArrowSchema) 
```